src/main/java/com/backend/liaison_system/users/admin/service/AdminServiceImpl.java

```diff
@@ -147,6 +148,11 @@ public Response<?> uploadStudents(MultipartFile file) {
                 //Ensure student does not already exist
                 Optional<Student> studentCheck = studentRepository.findByEmail(currentStudent.getEmail());
                 if(studentCheck.isEmpty()) {
```

Comment:
What happens when the incoming request is not a semester out; let's say it is empty?

I think you should look at that side too